python/cudf/cudf/tests/test_string.py

```diff
@@ -1775,14 +1775,23 @@ def test_string_count(data, pat, flags):
 
 
 def test_string_findall():
-    ps = pd.Series(["Lion", "Monkey", "Rabbit"])
-    gs = cudf.Series(["Lion", "Monkey", "Rabbit"])
+    test_data = ["Lion", "Monkey", "Rabbit", "Don\nkey"]
+    ps = pd.Series(test_data)
+    gs = cudf.Series(test_data)
 
     assert_eq(ps.str.findall("Monkey")[1][0], gs.str.findall("Monkey")[0][1])
```

Comment:
I noticed the indexing is different here. We should be asserting that the entire results are equal, not just the one value that matches the regex. However, it would probably fail right now: pandas returns a Series containing lists of strings, while cuDF returns a DataFrame. We should open an issue (if one doesn't exist) and adopt pandas' convention here -- but that's out of scope for this PR.

```python
>>> test_data = ["Lion", "Monkey", "Rabbit", "Don\nkey"]
>>> ps = pd.Series(test_data)
>>> gs = pd.Series(test_data)
>>> ps.str.findall("Monkey")
0          []
1    [Monkey]
2          []
3          []
dtype: object
>>> gs.str.findall("Monkey")
        0
0    <NA>
1  Monkey
2    <NA>
3    <NA>
```

Reply:
Yes, thanks. This is a bit beyond my Python skills.

Reply:
No worries. I'll file an issue and take a look sometime after this PR is merged. (I have been filing issues faster than I can resolve them lately so it may be a little bit.)

Reply:
If you are up to reviewing a similar PR, #10192 adds the regex-flags parameter to `extract`.

Reply:
Filed #10226.